spec/index.bs

```diff
@@ -1518,20 +1500,184 @@ enum SupportedType {
+            "`default-src`", then set |source-list| to that directive's [=directive/value=].
+
+        3.  If |source-list| is not `null`, and does not contain a [=source expression=] which is
+            an [=ASCII case-insensitive=] match for the string "<a grammar>`'unsafe-eval'`</a>",
+            then:
+
+            1.  Let |violation| be the result of executing [[CSP3#create-violation-for-global]] on
+                |global|, |policy|, and "`script-src`".
+
+            2.  Set |violation|'s [=violation/resource=] to "`inline`".
+
+            3.  If |source-list| [=list/contains=] the expression
+                "<a grammar>`'report-sample'`</a>", then set |violation|'s [=violation/sample=] to
+                the substring of <del>|source|</del> <ins>|sourceString|</ins> containing its first
+                40 characters.
+
+            4.  Execute [[CSP3#report-violation]] on |violation|.
+
+            5.  If |policy|'s [=policy/disposition=] is "`enforce`", then set |result| to
+                "`Blocked`".
+
+    3.  If |result| is "`Blocked`", throw an `EvalError` exception.
+
+10. <ins>Return |sourceString|.</ins>
+
+Note: returning |sourceString| means that the string that gets
+compiled is that returned by any [=default policy=] in the course of
+executing [$Get Trusted Type compliant string$].
+
+Issue: This depends on a solution to
+<a href="https://github.com/WICG/trusted-types/issues/144">issue #144</a> like <a
+href="https://mikesamuel.github.io/proposal-hostensurecancompilestrings-passthru/">TC39 HostBeforeCompile</a>
+
+Issue: In some cases, the violation "`'report-sample'`" contain the result of
+applying the default policy to a string argument which differs.
+Specifically when, there is a [=default policy=], |isExempt| is false,
+and |source| there is a CSP policy for either the |callerRealm|
+or |callerRealm| that disallows "`'unsafe-eval'"`.
+Is this a feature or a bug?
+
+Note: The previous algorithm reports violations via both report-uris where
+callerRealm != calleeRealm.  If [$Get Trusted Type compliant string$] reports an
+error, it only reports it via its |calleeRealm|'s report-uri.
+
+
+### <dfn abstract-op>IsSourceExempt</dfn> Algorithm ### {#is-source-exempt-algorithm}
+
+The IsSourceExempt algorithm takes a list of
+[[CSP3#content-security-policy-object|policies]] (|cspPolicies|),
+and a {{TrustedTypeConfiguration}} (|ttConfig|) and executes the
+following steps:
+
+1.  Let |isTTStrict| = `true`.
+1.  If |ttConfig|'s {{TrustedTypeConfiguration/unknownPolicyName}} is
+    {{UnknownPolicyNameDisposition|"allow"}} then
+    set |isTTStrict| to `false`.
+1.  Otherwise if |ttConfig|'s {{TrustedTypeConfiguration/allowedNames}} contains `'*'` then
+    set |isTTStrict| to `false`.
+1.  Let |isTTEnforced| = `false`.
+1.  If |ttConfig|'s {{TrustedTypeConfiguration/domSinks}} is
+    {{StringAtSinkDisposition|"reject"}} then
+    set |isTTEnforced| to `true`.
+1.  Let |isCSPEnforced| = `false`.
```

Comment:
The name might be a bit misleading, IIUC this is not checking the policy disposition, but whether any of the CSPs (enforcing, or report-only) cover `script-src` without `'unsafe-eval'`.

Reply:
Um.  isCSPEnforced should mean

∃ p. !isReportOnly(p)  and  'unsafe-eval' not in p

If not, that's a spec bug.
What do you think of the name given that goal?

Reply:
The name is fine (there's just no check for report only here, I think).